### ISSUE TITLE
Issue setting up Strategy [in TypeScript]

### ISSUE THREAD

**Issue by jackLIT** [superfaceai/passport-twitter-oauth2#25]
Hello, I am trying to setup an app to use this package but I am having issues setting the Strategy; specifically the following:
```
passport.use(
  new Strategy(
    {
      clientID: env.TWITTER_API_CLIENT_ID,
      clientSecret: env.TWITTER_API_CLIENT_SECRET,
      clientType: 'confidential',
      callbackURL: 'http://localhost:3000/auth/twitter/callback',
    },
    function (accessToken, refreshToken, profile, done) {
      return done(null, profile);
    }
  )
);
```

I have most commonly seen 2 errors between trying different things to resolve the issues:
1. 
```
Property 'authenticate' is missing in type 'import(".../node_modules/@superfaceai/passport-twitter-oauth2/dist/strategy").Strategy' but required in type 'import(".../node_modules/@types/passport/index").Strategy'.
```
2. 
```
'Strategy' cannot be used as a value because it was exported using 'export type'.
```

I will add that I am using typescript (not sure if that should be causing these issues though) and I am working off the example code [here](https://superface.ai/blog/twitter-oauth2-passport)

Thanks in advance for any help!!

**Comment by jnv** [comment]
Thanks for the report. It's possible the current CommonJS exports have some troubles with TypeScript since I focused on keeping the backwards compatibility of default + named export.

Could you please add a rest of your example? Notably, are you using `require` or `import` to import the strategy? And in the former case, did you try using `import`?

**Comment by Push-Stack** [comment]
I am working on a Nest JS project. 
I am getting the same error **Strategy' cannot be used as a value because it was exported using 'export type.**

`import { Strategy, Profile } from '@superfaceai/passport-twitter-oauth2';`

**Comment by janhalama** [comment]
I can confirm that there is an issue in TypeScript support. I have opened [PR](https://github.com/superfaceai/passport-twitter-oauth2/pull/26) with fixes.

**Comment by janhalama** [comment]
@jackLIT, @Push-Stack We've just released @superfaceai/passport-twitter-oauth2 v1.2.1 with TypeScript fixes.

**Comment by kazuemon** [comment]
I am using @superfaceai/passport-twitter-oauth2@1.2.2 and continue to have this problem.
Installing @types/passport-oauth2 at the same time solved the problem.

**Comment by jnv** [comment]
@kazuemon Thanks for the report. I will reopen this issue.

I see three options how to address it:

1. Add missing dependency on `@types/passport-oauth2`; easiest, but it has multiple transitive dependencies and it would add extra weight for non-TS users. 
2. Make `@types/passport-oauth2` an optional dependency and document it needs to be installed.
3. Pull in just the types we need from `@types/passport-oauth2` and its transitive dependencies.

Just that types package adds [3,5 MB install size](https://packagephobia.com/result?p=@types/passport-oauth2@1.4.11) which is why I'm not quite happy on slapping it on as an extra dependency.

**Comment by jnv** [comment]
I decided to go with `optionalDependencies` in #32. While it means the types get installed for non-TS users by default, it should be possible to omit them.

**Comment by jnv** [comment]
Hey folks, I've released [v1.2.3](https://github.com/superfaceai/passport-twitter-oauth2/releases/tag/v1.2.3) with optional `@types` dependencies today, so hopefully it should fix this issue for good.
